docs/madr/decisions/046-grpc-probes.md

```diff
@@ -0,0 +1,239 @@
+# gRPC probes
+
+- Status: 
+
+Technical Story: https://github.com/kumahq/kuma/issues/3875
+
+## Context and Problem Statement
+
+Kubernetes support three types of network based probes:
+- HTTP Get
+- TCP Socket
+- gRPC
+
+Kuma transparent proxy captures inbound traffic of pods, including the probe traffic. This leads to unexpected behaviour for Kubernetes probes without special handling: gRPC requests can fail due to lack of mTLS client certificates attached.
+
+We've implemented the [Virtual Probes feature](https://kuma.io/docs/2.6.x/policies/service-health-probes/#virtual-probes) which creates a dedicated listener for HTTP probes. And now, we want to support gRPC probes as well. We'll create a separated MADR for the supporting of TCP probes.
+
+Kubernetes users can specify HTTP and gRPC probes like this:
+
+```
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: app
+    startupProbe:
+      httpGet:
+        path: /healthz
+        port: 8080
+    livenessProbe:
+      grpc:
+        port: 2379
+        service: liveness
+```
+
+When a gRPC probe request is being sent from kubelet, it will:
+- include the port as part of the underlying HTTP `Host` header 
+- include the `service` as part of gRPC message body
+
+A pod can have multiple containers with each defines multiple and different probes, the worst case is a pod has all different types of probes with many probes for each type. 
```

Comment:
```suggestion
A pod can have multiple containers with each defining multiple and different probes, the worst case is a pod has all different types of probes with many probes for each type. 
```

Reply:
There's one detail though here as a port won't be used by different containers as there's a single netns for the whole pod.

Reply:
I meant defining different types (httpGet/gRPC/tcpSocket) of probes by:
- different probes (one on startupProe, the other on readinessProbe, etc.)
- on different containers